django/utils/html.py

```diff
@@ -317,18 +317,22 @@ def __call__(self, text, trim_url_limit=None, nofollow=False, autoescape=False):
         safe_input = isinstance(text, SafeData)
 
         words = self.word_split_re.split(str(text))
-        return "".join(
-            [
-                self.handle_word(
+        mapping = {}
+        urlized_words = []
+        for word in words:
+            if word in mapping:
+                urlized_words.append(mapping[word])
```

Comment:
Suggestion: If `mapping` could be quite a hot cache, perhaps worth having the "check" and "get" in the same statement?

```suggestion
            if urlized_word := mapping.get(word):
                urlized_words.append(urlized_word)
```

A bit of a micro-optimisation, but it's not much extra code here.